monitoring/rid_qualifier/create_flight_record_from_kml.py

```diff
@@ -0,0 +1,358 @@
+    Returns:
+        An altitude number.
+    """
+    distances = get_polygons_distances_from_point(point, polygons)
+    if min(distances) < 0.1:
+        # less than 1 meter, consider it almost on the polygon.
+        interpolated_value = all_possible_values[distances.index(min(distances))]
+    else:
+        dividend = 0
+        divisor = 0
+        for value, distance in zip(all_possible_values, distances):
+            dividend += value/distance
+            divisor += 1/distance 
+        interpolated_value = dividend / divisor
+    return round(interpolated_value, 2) if round_value else interpolated_value
+
+
+def get_speeds_from_speed_polygons(speed_polygons):
+    return [kml.get_polygon_speed(n) for n in list(speed_polygons)]
+   
+
+def get_flight_state_coordinates(flight_details):
+    """Returns flight's state coordinates at speed/sec.
+    Args:
+        flight_details: Flight details from the KML that include input coordinates.
+    Returns:
+        List of unflatten coordinates at a speed/sec interval and list of speeds at each point.
+    """
+    if flight_details.get('input_coordinates'):
+        input_coordinates = get_flight_coordinates(flight_details['input_coordinates'])
+    reference_point = input_coordinates[0] # TODO: Check `if input_coordinates:`.
+    
+    flatten_points = []
+    for point in input_coordinates:
+        flatten_points.append(flatten(
+            s2sphere.LatLng.from_degrees(*reference_point[:2]),
+            s2sphere.LatLng.from_degrees(*point[:2])
+        ))
+    
+    speed_polygons = flight_details['speed_polygons']
+    flattened_speed_polygons = get_flight_polygons_flattened(reference_point, speed_polygons)
+    all_polygon_speeds = get_speeds_from_speed_polygons(speed_polygons)
+    flight_state_vertices, flight_state_speeds = get_flight_state_vertices(
+        flatten_points, flattened_speed_polygons, all_polygon_speeds)
+    
+    alt_polygons = flight_details['alt_polygons']
+    flattened_alt_polygons = get_flight_polygons_flattened(reference_point, alt_polygons)
+    all_polygon_alts = [p[0][2] for p in list(alt_polygons.values())]
+
+    # return
+    
+    flight_state_altitudes = []
+    # flight_state_speeds = []
+    for point in flight_state_vertices:
+        flight_state_altitudes.append(
+            get_interpolated_value(point, flattened_alt_polygons, all_polygon_alts))
+        # flight_state_speeds.append(
+        #     get_interpolated_value(point, flattened_speed_polygons, all_polygon_speeds, round_value=True))
+    flight_state_vertices_unflatten = [unflatten(
+        s2sphere.LatLng.from_degrees(*reference_point[:2]), v) for v in flight_state_vertices]
+
+    # Position Lat, Lng to Lng, Lat order for KML representation.
+    flight_state_coordinates = []
+    for p, alt in zip(flight_state_vertices_unflatten, flight_state_altitudes):
+        flight_state_coordinates.append((
+            str(p.lng().degrees), str(p.lat().degrees), str(alt)
```

Comment:
We want the JSON values of `lat`, `lng`, and `alt` to be numbers rather than strings, so I think we want to drop the `str` calls on this line